jax/_src/scipy/special.py

```diff
@@ -2375,7 +2375,6 @@ def poch(z: ArrayLike, m: ArrayLike) -> Array:
   Notes:
     The JAX version supports only real-valued inputs.
   """
-  # Factorial definition when m is close to an integer, otherwise gamma definition.
```

Comment:
revert?

Reply:
This is a comment I forgot to remove when first implementing this func a while ago. This was reverted in the previous PR #18499 with [this comment](https://github.com/google/jax/pull/18499#discussion_r1393039344)

Reply:
Sounds good, thanks